src/app/features/reference-data/reference-data-add/reference-data-add.component.ts

```diff
@@ -77,13 +62,52 @@ export class AddReferenceDataComponent {
     return FormNodeWidth;
   }
 
-  get isFormValid(): boolean {
-    if (!this.form) return false;
-
-    const errors: GlobalError[] = [];
-    DynamicFormService.validate(this.form, errors);
-    this.globalErrorService.setErrors(errors);
-    return this.form.valid;
+  get template(): FormNode {
+    let templateToReturn: FormNode;
+    switch (this.type) {
+      case ReferenceDataResourceType.Brakes:
+        templateToReturn = brakesTemplate;
+        break;
+      case ReferenceDataResourceType.CountryOfRegistration:
+        templateToReturn = countryOfRegistrationTemplate;
+        break;
+      case ReferenceDataResourceType.HgvMake:
+        templateToReturn = hgvTemplate;
+        break;
+      case ReferenceDataResourceType.PsvMake:
+        templateToReturn = psvTemplate;
+        break;
+      case ReferenceDataResourceType.ReasonsForAbandoningHgv:
+        templateToReturn = reasonsForAbandoningHgvTemplate;
+        break;
+      case ReferenceDataResourceType.ReasonsForAbandoningPsv:
+        templateToReturn = reasonsForAbandoningPsvTemplate;
+        break;
+      case ReferenceDataResourceType.ReasonsForAbandoningTrl:
+        templateToReturn = reasonsForAbandoningTrlTemplate;
+        break;
+      case ReferenceDataResourceType.SpecialistReasonsForAbandoning:
+        templateToReturn = specialistReasonsForAbandoningTemplate;
+        break;
+      case ReferenceDataResourceType.TirReasonsForAbandoning:
+        templateToReturn = reasonsForAbandoningTirTemplate;
+        break;
+      case ReferenceDataResourceType.TrlMake:
+        templateToReturn = trlTemplate;
+        break;
+      case ReferenceDataResourceType.Tyres:
+        templateToReturn = tyresTemplate;
+        break;
+      default:
+        templateToReturn = {} as FormNode;
+        break;
+    }
+    templateToReturn.children?.forEach(child => {
+      if (child.name === 'resourceKey') {
+        child.disabled = false;
+      }
+    });
+    return templateToReturn;
```

Comment:
I'm not sure why you need to do that if the control is disabled in the controls anyways. If you do that, you can also get rid of the `let` and the `break`s, but just returning inside of the switch statement.